src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2436,13 +2496,18 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                     manifestInfo.AddToModuleList(m);
                 }
             }
+
             if (compatiblePSEditions != null)
             {
-                foreach (var psEdition in compatiblePSEditions)
-                {
-                    manifestInfo.AddToCompatiblePSEditions(psEdition);
-                }
+                manifestInfo.AddToCompatiblePSEditions(compatiblePSEditions);
             }
+
+// Modules loaded from the System32 module path may
+// not have the CompatiblePSEditions fields ignored
+#if !UNIX
+            manifestInfo.IsLoadedFromCompatibilityCheckedPath = isOnSystem32ModulePath;
+#endif
```

Comment:
I think it's not needed to put this in `if/def` here. On Unix plat, `isOnSystem32ModulePath` will always be false, and it's fine to set it here to keep a single code path.